pkg/remoteclient/informers/factory_test.go

```diff
@@ -272,7 +273,10 @@ func TestDynamicSharedInformerFactory_EventHandles_MultipleClusters(t *testing.T
 			t.Error(err)
 		}
 
-		observedObjects := drainChannel(ctx, informerUpdateObjCh, time.Second)
+		// Set drainingTimeout to be a bit less than the total test timeout divided by the number of test cases to ensure
+		// we still have time left for other operations to complete.
+		drainingTimeout := time.Duration(int(testTimeout)/len(tss)) - (time.Millisecond * 500)
```

Comment:
I've got a feeling that leaving 500ms for 6 updates, 6 waits for cache sync and entire test setup is going to introduce more flakiness. Lets leave more space for it.

Reply:
Sounds reasonable. Bumped the total test timeout to 20s instead of 10, and ensured we have 10s for other operations spared.